File: HelloWorld/object_oriented.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import sys

class Greeter :

  def __init__( self, name = 'World' ) :
    self.name = name

  def say( self ) :
    return self.name

if __name__ == "__main__" :

  if len( sys.argv ) - 1 == 0 :
    greeter = Greeter()
  else :
    greeter = Greeter( sys.argv[ 1 ] )
  print( "Hello, " + greeter.say() + "!" )

  print( "» José Carlos Monteiro «" )

File: HelloWorld/object_oriented.go
package main

import (
  "fmt"
  "os"
)

const message string = "Hello, %s!\n"

type Greeter struct {
  name string
}

func ( g * Greeter ) say() string {
  return g.name
}

func main() {
  var greeter Greeter
  if len( os.Args[ 1: ] ) == 0 {
    greeter = Greeter{ name: "World" }
  } else {
    greeter = Greeter{ name: os.Args[ 1 ] }
  }
  fmt.Printf( message, greeter.say()  )

  fmt.Println( "» José Carlos Monteiro «" )
}
